### ISSUE TITLE
JOIN isn't working when Align feature is on

### ISSUE THREAD

**Issue by peheje** [mechatroner/vscode_rainbow_csv#37]
I can't get JOIN RBQL to work when the two files I'm joining have been Aligned with spaces. After Shrink it works again.

**Comment by mechatroner** [comment]
Originally I though that "Align" mode would be used only for viewing purposes. And JOIN doesn't work because it compares literal string values of fields.
I guess it is possible to always strip whitespaces from fields in JOIN (and all other?) RBQL queries maybe with a warning that they were stripped.

**Comment by mechatroner** [comment]
Another possible solution: never strip whitespaces, but detect them, and if whitespaces were found, show a warning at the end that "Some fields have trailing whitespaces this might affect query result set: RBQL uses literal string comparison by default". So in your case JOIN still won't work but you would get the explanatory warning.

**Comment by peheje** [comment]
Thanks for the response mechatroner

A warning would most accurate, because the user might _want_ whitespaces to be part of the data, i.e.

` "a" !== " a" `

As you suggested: the warning could be displayed when whitespaces are detected in the columns mentioned in the "ON" query. 
The warning could suggest shrinking the files beforehand, but the user shall do this himself.

**Comment by mechatroner** [comment]
I agree, I also like this "never-discard-whitespaces" policy more than "discard-with-warning" option. Besides Item 4 in CSV rfc recommends to keep them: https://tools.ietf.org/html/rfc4180

I also found that current handling of trailing and leading whitespace in the RBQL code is not consistent for CSV files (TSV should be OK though) 😞  I will try to fix this ASAP.

And another thing to consider about keeping/discarding whitespaces, consider the following alligned line:
`  foo  ,  bar   ,    "   hello ""world""   "   , foobar  `

The data payload in field 3 is probably ` hello "world" ` without external whitespaces outside double quotes and double quotes. Basically to extract payload from quoted fields we remove the external double quotes and external whitespaces (but keep trailing spaces inside doblequotes). But to preserve consistency we should also strip alignment whitespaces from fields 1, 2, and 4 which would contradict our "never-discard-whitespaces" policy.

Another solution is to interpret data payload in field 3 in my example differently: just remove double quotes and keep all whitespaces even those outside double quotes. I guess this is a better option.
